ocsp_updater/updater.go

```diff
@@ -330,21 +335,31 @@ func (updater *OCSPUpdater) generateResponse(ctx context.Context, status sa.Cert
 
 // storeResponse stores a given CertificateStatus in the database.
 func (updater *OCSPUpdater) storeResponse(ctx context.Context, status *sa.CertStatusMetadata) error {
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
+	// If a redis client is configured, try to store the response in redis.
 	if updater.rocspClient != nil {
+		// Create a context to set a deadline for the goroutine that stores
+		// the response in redis. Set the timeout to one second longer than
+		// the configured redis timeout to give redis a chance to return a
+		// timeout error first. This context is necessary because we don't
+		// want to wait to confirm a write to redis (best effort), which
+		// causes a race with the Tick() context cancellation if the parent
+		// context is used. When writing to redis is the primary storage
+		// source we can change to use the parent context.
+		ctx2, cancel := context.WithTimeout(context.Background(), updater.redisTimeout+time.Second)
```

Comment:
1. I think `ctx`, declared on line `337`, is no longer being used. You could just remove that declaration and rename `ctx2` to `ctx`. 
1. Alternatively, I think this context declaration could be moved up to `464`, just before `storeResponse()` is called.

Reply:
Thanks!

1. Correct, it is no longer currently being used. It is probably cleanest to get rid of it, but it is currently satisfying an interface, so I figured not to mess with it because we will want to use it in the near future when redis is no longer a best effort to store.
2. If I move the creation of the context outside of the function (or even outside of the if statement) I will need to `cancel()` on all possible paths. In that case I would `defer cancel()` inside the goroutine same as now, and then also as an `else` alternative to the `if` on line `339` to account for all paths and to avoid context leaks. It is certainly an option, just seemed a degree higher of "icky feeling" to me.